tools/cli/commands/build.js

```diff
@@ -524,6 +524,40 @@ async function copyFileAtomic( src, dest ) {
 	}
 }
 
+/**
+ * Check for filename collisions.
+ *
+ * @param {string} basedir - Base directory.
+ * @returns {string[]} Colliding file names.
+ */
+async function checkCollisions( basedir ) {
+	// @todo Once we require Node 20.1+, use the new `recursive` option to `fs.readdir` instead of manually recursing here.
+	// Doing `const files = await fs.readdir( basedir, { recursive: true } );` should suffice.
+	const files = [];
+	const ls = async dir => {
+		for ( const file of await fs.readdir( dir, { withFileTypes: true } ) ) {
+			const path = npath.join( dir, file.name );
+			files.push( npath.relative( basedir, path ) );
+			if ( file.isDirectory() ) {
+				await ls( path );
```

Comment:
What if there are other folders in that path too? How would we recurse into them?